pkg/archive/paths.go

```diff
@@ -104,4 +142,20 @@ func WalkTarGzFiles(ctx context.Context, path string, walkFn FileWalkFn) (err er
 		}
 	}
 	return nil
+
+}
+
+func CheckArchiveType(filename string) FormatType {
+	fileSplit := strings.Split(filename, ".")
+	if len(fileSplit) < 2 {
+		return UnsupportedArchive
+	}
+
+	for i := len(fileSplit) - 1; i > 0; i-- {
```

Comment:
We know that we will only ever need to check two of the segments, so we can limit the number of times this loops through.